### PR TITLE
Use latest Debian version, install icinga and modules from packages.

### PR DESCRIPTION
**Pull request by rndmh3ro**

### CODE REVIEW

File: content/opt/setup/20-mysql [hunk 1]
```diff
@@ -9,12 +9,12 @@
 } || exit 0
 
 chown -R mysql:mysql /var/lib/mysql
-chown -R mysql:mysql /var/log/mysql
+chmod +x /var/ /var/lib/ /var/lib/mysql/
```
Comment (schurzi) [inline content/opt/setup/20-mysql:12]: x-bit missing on existing directories? How is that possible?

File: content/opt/setup/50-icinga2 [hunk 2]
```diff
@@ -24,13 +24,26 @@ mkdir -p /var/log/icinga2/compat/archives
...
+chown -R nagios:nagios /usr/share/icinga*
+chmod +x /var/log/icinga2/
+chmod +x /var/lib/icinga2/certs
+chmod +x /etc/
+chmod +x /etc/icinga2/
+chmod +x /usr
+chmod +x /usr/share
+chmod +x /usr/share/icinga
+chmod +x /usr/share/icinga2
+chmod +x /usr/share/icinga2/include
```
Comment (schurzi) [inline content/opt/setup/50-icinga2]: also here. maybe there is a wrong umask?

File: content/opt/setup/61-icingaweb2-director [hunk 3]
```diff
@@ -16,6 +16,7 @@ mysql_wait \
 
 if evaluate_boolean "${ICINGA2_FEATURE_DIRECTOR}"; then
 
+  ln -fs /usr/share/icingaweb2/modules/director /etc/icingaweb2/enabledModules/
 	icingacli module enable director
```
Comment (schurzi) [inline content/opt/setup/61-icingaweb2-director]: these two lines are doing the same thing, or are there differences?

File: Dockerfile [hunk 4]
```diff
@@ -131,17 +104,24 @@ RUN true \
     && mkdir -p /etc/icinga2 \
     && usermod -aG icingaweb2 www-data \
     && usermod -aG nagios www-data \
+    && usermod -aG icingaweb2 nagios \
```
Comment (schurzi) [inline Dockerfile:105]: could be merged with line 101

File: Dockerfile [hunk 5]
```diff
@@ -14,22 +14,27 @@ ENV APACHE2_HTTP=REDIRECT \
...
+    ICINGA2_LOG_LEVEL="information" \
     MYSQL_ROOT_USER=root
 
 RUN export DEBIAN_FRONTEND=noninteractive \
     && apt-get update \
     && apt-get upgrade -y \
     && apt-get install -y --no-install-recommends \
     apache2 \
+    apt-transport-https \
+    bc \
```
Comment (schurzi) [inline Dockerfile]: why do we need a calculator here? :D

File: content/opt/setup/20-mysql (hunk 1 above)
Comment (rndmh3ro) [reply content/opt/setup/20-mysql:12]: It took some time to find the reason. :D
We do a `COPY content/ /`. When doing this, we copy `content/var` to the container. The directory-permissions from the local file are copied as-is to the container.
So if I have this locally:

```> ls -lsah content/
total 24K
4.0K drwxr-x--- 6 segu domänen-benutzer 4.0K Aug 24  2022 .
4.0K drwxr-x--- 7 segu domänen-benutzer 4.0K Nov 14 10:26 ..
4.0K drwxr-x--- 5 segu domänen-benutzer 4.0K Aug 24  2022 etc
4.0K drwxr-x--- 4 segu domänen-benutzer 4.0K Aug 24  2022 opt
4.0K drwxr-x--- 3 segu domänen-benutzer 4.0K Aug 24  2022 usr
4.0K drwxr-x--- 3 segu domänen-benutzer 4.0K Aug 24  2022 var
```

then the other-group in the container won't have executable-permissions, too.

If you have a umask `027` locally, then when checking out the repository the directories will look like above. Then when copied into the container, the build will fail.
I also cannot add the directory-permissions to git, this seems to only work with files. That's why we need the chmod in the build.

File: content/opt/setup/61-icingaweb2-director (hunk 3 above)
Comment (rndmh3ro) [reply content/opt/setup/61-icingaweb2-director]: You're right, I removed one.

File: Dockerfile (hunk 5 above)
Comment (rndmh3ro) [reply Dockerfile]: It's a dependency:

```
root@d9c38dcf6a43:/# aptitude why bc
i   nagios-plugins-contrib     Depends    monitoring-plugins-contrib
i A monitoring-plugins-contrib Recommends bc
```

I removed the explicit install.

File: content/opt/setup/20-mysql (hunk 1 above)
Comment (schurzi) [reply content/opt/setup/20-mysql:12]: awesome :D ... not

**Review by schurzi — APPROVED** [review]

**Pull request merged**